_appmap/test/test_test_frameworks.py

```diff
@@ -4,83 +4,86 @@
 import json
 import re
 import sys
+from abc import ABC, abstractmethod
 
 import pytest
 
-from appmap.test.helpers import DictIncluding
+from _appmap import web_framework
 
-from .._implementation import web_framework
+from ..test.helpers import DictIncluding
 from .normalize import normalize_appmap
 
 
-def test_unittest_runner(testdir):
-    testdir.run(sys.executable, "-m", "unittest", "-vv")
+class _TestTestRunner(ABC):
+    @abstractmethod
+    def run_tests(self, testdir):
+        pass
 
-    assert len(list(testdir.output().iterdir())) == 7
-    verify_expected_appmap(testdir)
-    verify_expected_metadata(testdir)
+    def test_disabled(self, testdir, monkeypatch):
+        monkeypatch.setenv("APPMAP", "false")
 
+        self.run_tests(testdir)
 
-def test_appmap_unittest_runner(testdir):
-    testdir.run(sys.executable, "-m", "appmap.unittest", "-vv")
+        assert not testdir.output().exists()
 
-    verify_expected_appmap(testdir)
-    verify_expected_metadata(testdir)
 
+class TestUnittestRunner(_TestTestRunner):
+    def run_tests(self, testdir):
+        testdir.run(sys.executable, "-m", "unittest", "-vv")
 
-def test_appmap_unittest_runner_disabled(testdir, monkeypatch):
-    # since not setting APPMAP records by default, must disable it explicitly
-    monkeypatch.setenv("APPMAP", "false")
+    def test_enabled(self, testdir):
+        self.run_tests(testdir)
 
-    result = testdir.run(
-        sys.executable,
-        "-m",
-        "appmap.unittest",
-        "-vv",
-        "simple.test_simple.UnitTestTest.test_hello_world",
-    )
-    assert result.ret == 0
-    r = re.compile(r"AppMap disabled")
-    # when APPMAP=false it no longer prints a warning
-    assert [l for l in filter(r.search, result.errlines)] == []
```

Comment:
This test was deleted because using `appmap.unittest` as a test runner is no longer supported (I'm not sure it made sense to begin with).

I'll add tests for `APPMAP_RECORD_[PYTEST|UNITTEST]=false`. Setting those environment variables to `true` isn't supported.

Reply:
Should also update the documentation to not refer to `true` for these variables under ["Advanced Configuration"](https://appmap.io/docs/reference/appmap-python.html).

Reply:
This part I have done: https://github.com/getappmap/applandinc.github.io/pull/931 .